From 969526c029ee10d8201bc75b110daaa0e67f358b Mon Sep 17 00:00:00 2001
From: Jonathan DE CESCO <jonathan.de.cesco@intel.com>
Date: Thu, 8 Nov 2012 15:54:09 +0100
Subject: [PORT FROM MAIN] wifi: display allowed multicast address list

BZ: 66460

To ease wifi filtering debugging, it is convenient to display
the list of multicast addresses from whose packets will wake
up the host.
In conjunction with iw wowlan show, we can get a complete picture
of packet filtering configuration at any time

Change-Id: I697e70037a34bc161e222bea6fbeb490878688ca
Orig-Change-Id: Iaa8eb2c95fefd2286fba4831945bce797b0ca0ad
Signed-off-by: Jonathan DE CESCO <jonathan.de.cesco@intel.com>
Reviewed-on: http://android.intel.com:8080/73237
Reviewed-by: Casasnovas, QuentinX <quentinx.casasnovas@intel.com>
Reviewed-by: Isaksen, FrodeX <frodex.isaksen@intel.com>
Tested-by: Champciaux, Nicolas <nicolas.champciaux@intel.com>
Reviewed-by: buildbot <buildbot@intel.com>
Tested-by: buildbot <buildbot@intel.com>
Reviewed-on: http://android.intel.com:8080/78004
Reviewed-by: Li, XiaojingX <xiaojingx.li@intel.com>
Tested-by: Li, XiaojingX <xiaojingx.li@intel.com>
---
 wl12xx-compat/drivers/net/wireless/wl12xx/main.c |    6 ++++++
 1 files changed, 6 insertions(+), 0 deletions(-)

diff --git a/wl12xx-compat/drivers/net/wireless/wl12xx/main.c b/wl12xx-compat/drivers/net/wireless/wl12xx/main.c
index 43cc17b..66633ad 100644
--- a/wl12xx-compat/drivers/net/wireless/wl12xx/main.c
+++ b/wl12xx-compat/drivers/net/wireless/wl12xx/main.c
@@ -3420,6 +3420,7 @@ static u64 wl1271_op_prepare_multicast(struct ieee80211_hw *hw, int mc_count,
 		fp->enabled = false;
 #if (LINUX_VERSION_CODE >= KERNEL_VERSION(2,6,35))
 	} else {
+		wl1271_notice("Allowed multicast addresses:");
 		fp->enabled = true;
 		netdev_hw_addr_list_for_each(ha, mc_list) {
 #else
@@ -3435,6 +3436,11 @@ static u64 wl1271_op_prepare_multicast(struct ieee80211_hw *hw, int mc_count,
 #else
 			       mc_list->da_addr, ETH_ALEN);
 #endif
+			wl1271_notice("[%02x:%02x:%02x:%02x:%02x:%02x]", fp->mc_list[fp->mc_list_length][0],
+				fp->mc_list[fp->mc_list_length][1], fp->mc_list[fp->mc_list_length][2],
+				fp->mc_list[fp->mc_list_length][3], fp->mc_list[fp->mc_list_length][4],
+				fp->mc_list[fp->mc_list_length][5]);
+
 			fp->mc_list_length++;
 #if (LINUX_VERSION_CODE >= KERNEL_VERSION(2,6,35))
 		}
-- 
1.7.4.1

